### PR TITLE
Do not put switch into inactive state on network start

### PR DESCRIPTION
**Pull request by timofei-durakov**

### CODE REVIEW

**Review by dpoltavets — APPROVED** [review]

File: src-java/network-topology/network-storm-topology/src/main/java/org/openkilda/wfm/topology/network/service/NetworkSwitchService.java [hunk 1]
```diff
@@ -72,7 +73,11 @@ public void switchAddWithHistory(HistoryFacts history) {
                 .history(history)
                 .build();
         controller.put(history.getSwitchId(), switchFsm);
-        controllerExecutor.fire(switchFsm, SwitchFsmEvent.HISTORY, fsmContext);
+        SwitchFsmEvent event = SwitchFsmEvent.HISTORY_DOWN;
+        if (history.getLastRecordedStatus().equals(SwitchStatus.ACTIVE)) {
```
Comment (niksv) [inline src-java/network-topology/network-storm-topology/src/main/java/org/openkilda/wfm/topology/network/service/NetworkSwitchService.java:77]: I suggest to use
```
SwitchStatus.ACTIVE.equals(history.getLastRecordedStatus())
```
to do not face with NPE

**Review by niksv — APPROVED** [review]

**Review by dpoltavets — APPROVED** [review]

**Review by niksv — APPROVED** [review]

**Review by dpoltavets — APPROVED** [review]

File: src-java/network-topology/network-storm-topology/src/main/java/org/openkilda/wfm/topology/network/controller/sw/SwitchFsm.java [hunk 2]
```diff
@@ -483,10 +491,14 @@ private boolean isPhysicalPort(int portNumber) {
                     PersistenceManager.class, SwitchId.class, NetworkOptions.class);
 
             // INIT
-            builder.transition()
-                    .from(SwitchFsmState.INIT).to(SwitchFsmState.OFFLINE).on(SwitchFsmEvent.HISTORY)
-                    .callMethod("applyHistory");
             builder.transition().from(SwitchFsmState.INIT).to(SwitchFsmState.SYNC).on(SwitchFsmEvent.ONLINE);
+            builder.transition().from(SwitchFsmState.INIT).to(SwitchFsmState.PENDING).on(SwitchFsmEvent.HISTORY);
+
+            // DB_ONLINE
```
Comment (surabujin) [inline src-java/network-topology/network-storm-topology/src/main/java/org/openkilda/wfm/topology/network/controller/sw/SwitchFsm.java:497]: old name

File: docs/design/network-discovery/switch-FSM.puml [hunk 3]
```diff
@@ -28,8 +28,12 @@ endlegend
 
 [*] --> INIT
 
-INIT --> OFFLINE : history / setup port workers
 INIT --> SYNC : online
+INIT --> PENDING : history / setup port workers
```
Comment (surabujin) [inline docs/design/network-discovery/switch-FSM.puml:32]: "setup port worker" is part of "apply history"

**Review by surabujin — APPROVED** [review]

File: docs/design/network-discovery/switch-FSM.puml (hunk 3 above)
Comment (timofei-durakov) [reply docs/design/network-discovery/switch-FSM.puml]: changed, thanks

**Review by niksv — APPROVED** [review]

**Review by surabujin — APPROVED** [review]

**Review by rozdy — APPROVED** [review]

**Pull request merged**